activesupport/lib/active_support/message_encryptor.rb

```diff
@@ -43,65 +50,181 @@ class InvalidMessage < StandardError; end
     # * <tt>:digest</tt> - String of digest to use for signing. Default is +SHA1+.
     # * <tt>:serializer</tt> - Object serializer to use. Default is +Marshal+.
     def initialize(secret, *signature_key_or_options)
-      options = signature_key_or_options.extract_options!
-      sign_secret = signature_key_or_options.first
+      @options = signature_key_or_options.extract_options!
       @secret = secret
-      @sign_secret = sign_secret
-      @cipher = options[:cipher] || 'aes-256-cbc'
```

Comment:
Imagine a user upgrading to Rails 5, where they call `MessageEncryptor.new(@secret, cipher: 'aes-128-gcm')`. This won't work because there's no cipher option anymore.